crates/languages/src/python.rs

```diff
@@ -427,6 +434,100 @@ impl ToolchainLister for PythonToolchainProvider {
     }
 }
 
+pub struct EnvironmentApi<'a> {
+    global_search_locations: Option<Arc<Mutex<Vec<PathBuf>>>>,
+    project_env: &'a HashMap<String, String>,
+    pet_env: pet_core::os_environment::EnvironmentApi,
+}
+
+impl<'a> EnvironmentApi<'a> {
+    pub fn from_env(project_env: &'a HashMap<String, String>) -> Self {
+        let paths = project_env
+            .get("PATH")
+            .map(|p| std::env::split_paths(p).collect());
+
+        EnvironmentApi {
+            global_search_locations: paths.map(|p| Arc::new(Mutex::new(p))),
+            project_env,
+            pet_env: pet_core::os_environment::EnvironmentApi::new(),
+        }
+    }
+
+    fn user_home(&self) -> Option<PathBuf> {
+        self.project_env
+            .get("HOME")
+            .or_else(|| self.project_env.get("USERPROFILE").cloned())
+            .map(|home| pet_fs::path::norm_case(PathBuf::from(home)))
+            .or_else(self.pet_env.get_user_home())
+    }
+}
+
+#[cfg(windows)]
+impl<'a> pet_core::os_environment::Environment for EnvironmentApi<'a> {
```

Comment:
These two cfgs differ just by `get_know_global_search_locations`, so ig we could #[cfg] at that functions level (have two different impls for that function and unify the rest).

Reply:
Yeah, I've just unified the implementations